layouts/shortcodes/upcoming-releases.html

```diff
@@ -12,13 +12,21 @@
       .Site.Data.releases.schedule.upcoming_releases }}
       <tr>
         <td>
-          {{ time.Format ( T "release_date_format_month") $patchReleaseInfo.targetDate }}
+          {{- with $patchReleaseInfo.targetDate -}}
+          {{ time.Format ( T "release_date_format_month" ) . }}
+          {{- end -}}
         </td>
+        {{- with $patchReleaseInfo.cherryPickDeadline -}}
         <td>
-          {{ time.Format ( T "release_date_format") $patchReleaseInfo.cherryPickDeadline }}
+          <time title="{{ time.Format ( T "release_date_format_text") . }}" datetime={{ time.Format "2006-01-02" . }}>{{ time.Format ( T "release_date_format" ) . }}</time>
         </td>
+        {{- else -}}
+          <td class="value-not-applicable">-</td>
+        {{- end -}}
         <td>
-          {{ time.Format ( T "release_date_format") $patchReleaseInfo.targetDate }}
+          {{- with $patchReleaseInfo.targetDate -}}
+          <time title="{{ time.Format ( T "release_date_format_text") . }}" datetime={{ time.Format "2006-01-02" . }}>{{ time.Format ( T "release_date_format" ) . }}</time>
```

Comment:
I'm concerned with the direction of this change is leading us to.
There is no doubt that the automated generation of certain pages or elements could help improve productivity. But this is still a documentation project anyway. Over engineering may not be a good thing to website maintainers. Whenever possible, we should try "keep it simple and stupid (aka KISS)". Carving on shit is not something we as a community should encourage.

Reply:
- The patch releases page is one of our most frequently updated pages and largely changes through data-driven changes.
- Rather than improve productivity for SIG Release, my intent here is to improve UX for people reading the rendered page.
  For example, if you can see that an EOL date is a date your business does not usually work, that helps you plan.
  It's even better for firms not to wait until the last possible moment to make a change, but I know there are corporations that wait exactly that long.

Reply:
I suggested an alternative, _more_ complicated approach: https://github.com/kubernetes/website/issues/46426